starter-code/src/viking.js

```diff
@@ -1,11 +1,66 @@
 // Soldier
-function Soldier() {}
+function Soldier (health, strength ) {
+    this.health = health;
+    this.strength = strength;
+     };
+Soldier.prototype.attack= function(){
+    return this.strength;
+}
+Soldier.prototype.receiveDamage= function(damage){
 
+    this.health -= damage;
+}
 // Viking
-function Viking() {}
+Viking.prototype=Object.create(Soldier.prototype);
+function Viking ( name, strenght, health) {
+    this.name  = "Harald";
+    Soldier.call(this, strenght, health);
+}
```

Comment:
Ici tu attribues à this.name la valeur "Harald". Du coup quoi qu'il arrive, tous tes vikings s'appelleront toujours Harald, ce qui n'est pas ce qu'on veut ;)
Comme pour le constructor de Soldier, l'idéal est d'assigner à this.name la valeur "name", qui est passé en argument de la fonction. Tu te retrouves donc avec :
```javascript
function Viking ( name, strenght, health) {
+    this.name  = name;
+    Soldier.call(this, strenght, health);
+}
```

Ce qui fait que quand on créera nos vikings, c'est la valeur passée à la place de "name" qui deviendra this.name, et on pourra choisir des noms différents pour tous nos guerriers.
Niveau organisation c'est plus pratique ;)